lib/tasks/test.js

```diff
@@ -60,17 +82,115 @@ module.exports.nonSilentCompilation = function(gulp) {
+			commandLine.run('npm', ['test'])
+				.then(function(output) {
+					log.primary('Running "npm test"...');
+					log.secondary(output.stdout);
+				}, function(output) {
+					log.primary('Running "npm test"...');
+					log.secondary(output.stdout);
+					log.secondaryError(output.stderr);
+					process.exit(1);
+				});
+		}
+	});
+};
+
+module.exports._runBrowserTest = function(config) {
+	return new Promise(function(resolve, reject) {
+		var nightwatchCommand = path.resolve('.', 'node_modules/nightwatch/bin/nightwatch');
+		var nightwatchConfigPath = config.nightwatchConfig || path.resolve('.', 'test/browser/nightwatch.json');
+		fs.readFile(nightwatchConfigPath, function(err, configContent) {
+			if (err) {
+				reject(err);
+			}
+
+			configContent = JSON.parse(configContent);
+			configContent.test_settings.default.launch_url = config.testUrl;
+
+			fs.writeFile(nightwatchConfigPath, JSON.stringify(configContent), function(error) {
+				if (error) {
+					reject(error);
+				}
+
+				resolve(commandLine.run(nightwatchCommand, ['--group', 'test/browser/tests',
+						'--config', nightwatchConfigPath,
+						'--env', ' chrome37_Grid,chrome38_Grid,chrome39_Grid,chrome40_Grid,firefox30_Grid,firefox31_Grid,firefox32_Grid,firefox33_Grid,firefox34_Grid,firefox35_Grid,ie8_Grid,ie9_Grid,ie10_Grid,ie11_Grid,safari7_Grid'
+				]));
 			});
+		});
+	});
+};
+
+module.exports.browserTest = function(gulp, config) {
+	config = config || {};
+	var appName = files.getModuleName();
+	var token;
+	var commit;
+	haikro.logger.setLevel('debug');
+
+	function cleanupTest(output) {
+		log.secondary(output.stdout);
+		log.secondaryError(output.stderr);
+		fs.unlink('Procfile');
+		return haikro.destroy({
+			token: token,
+			app: appName
+		});
 	}
+
+	return Promise.all([
+		process.env.HEROKU_AUTH_TOKEN ? Promise.resolve(process.env.HEROKU_AUTH_TOKEN) : commandLine.run('heroku', ['auth:token']),
+		commandLine.run('git', ['rev-parse HEAD']),
+		commandLine.run('npm', ['install', 'http-server']),
+		demo(gulp, { local: true })
+	])
+	.then(function(results) {
+		token = results[0].stdout ? results[0].stdout.trim() : results[0].trim();
+		commit = results[1].stdout.trim();
+		appName = appName + commit.substr(0, 5);
+		return haikro.create({
+			app: appName,
+			region: 'eu',
+			// organization: 'financial-times',
+			token: token
+		});
+	})
+	.then(function() {
+		fs.writeFileSync('Procfile', 'web: http-server demos/local', 'utf8');
+		return haikro.build({
+			project: process.cwd()
+		});
+	})
+	.then(function() {
+		return haikro.deploy({
+			app: appName,
+			token: token,
+			project: process.cwd(),
+			commit: commit
+		});
+	})
+	.then(function() {
+		config.testUrl = config.testUrl || 'https://' + appName + '.herokuapp.com';
+		return module.exports._runBrowserTest(config);
+	})
+	.then(function(output) {
+		cleanupTest(output);
+	})
+	.catch(function(e) {
+		if (e.stdout) {
+			cleanupTest(e)
+				.then(function() {
+					process.exit(1);
```

Comment:
:( `process.exit` makes me sad.  We should think about how we can have a cleaner interface in the future.